build.psm1

```diff
@@ -2508,6 +2528,95 @@ function Clear-PSRepo
     }
 }
 
+# Install PowerShell modules such as PackageManagement, PowerShellGet
+function Restore-PSModule
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Name,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Destination,
+
+        [string]$SourceLocation="https://powershell.myget.org/F/powershellmodule/api/v2",
+
+        [string]$RequiredVersion
+        )
+
+    $needRegister = $true
+    $RepositoryName = "mygetpsmodule"
+
+    # Check if the PackageManagement works in the base-oS or PowerShellCore
+    Get-PackageProvider -Name NuGet -ForceBootstrap -Verbose:$VerbosePreference
+
+    # Get the existing registered PowerShellGet repositories
+    $psrepos = PowerShellGet\Get-PSRepository
+
+    foreach ($repo in $psrepos)
+    {
+        if(($repo.SourceLocation -eq $SourceLocation) -or ($repo.SourceLocation.TrimEnd("/") -eq $SourceLocation.TrimEnd("/")))
+        {
+            # found a registered repository that matches the source location
+            $needRegister = $false
+            $RepositoryName = $repo.Name
+            break
+        }
+    }
+
+    if($needRegister)
+    {
+        $regVar = PowerShellGet\Get-PSRepository -Name $RepositoryName -ErrorAction SilentlyContinue
+        if($regVar)
+        {
+            PowerShellGet\UnRegister-PSRepository -Name $RepositoryName
+        }
+
+        log "Registering PSRepository with name: $RepositoryName and sourcelocation: $SourceLocation"
+        PowerShellGet\Register-PSRepository -Name $RepositoryName -SourceLocation $SourceLocation -ErrorVariable ev -verbose
+        if($ev)
+        {
+            throw ("Failed to register repository '{0}'" -f $RepositoryName)
+        }
+
+        $regVar = PowerShellGet\Get-PSRepository -Name $RepositoryName
+        if(-not $regVar)
+        {
+            throw ("'{0}' is not registered" -f $RepositoryName)
+        }
+    }
+
+    log ("Name='{0}', Destination='{1}', Repository='{2}'" -f $Name, $Destination, $RepositoryName)
+
+    $command = @{
+                    Name=$Name
+                    Path = $Destination
+                    Repository =$RepositoryName
+                }
+
+    if($RequiredVersion)
+    {
+        $command.Add("RequiredVersion", $RequiredVersion)
+    }
+
+    # pull down the module
+    log "running save-module $Name"
+    PowerShellGet\Save-Module @command -Force -Verbose
+
+    # Clean up
+    if($needRegister)
+    {
+        $regVar = PowerShellGet\Get-PSRepository -Name $RepositoryName -ErrorAction SilentlyContinue
+        if($regVar)
+        {
+            log "Unregistering PSRepository with name: $RepositoryName"
+            PowerShellGet\UnRegister-PSRepository -Name $RepositoryName
+        }
```

Comment:
can be simplified with below
            log "Unregistering PSRepository with name: $RepositoryName"
            PowerShellGet\Get-PSRepository -Name $RepositoryName -ErrorAction SilentlyContinue | PowerShellGet\UnRegister-PSRepository -Name $RepositoryName